### PR TITLE
BREAKING CHANGE: switch to ESM and replace superagent with native fetch

### PR DESCRIPTION
**Pull request by dlockhart**
Background: We'd like to purge `superagent` from our dependency chain in BSI, as it depends on older versions of libraries that are in conflict with other things in BSI. Also, `superagent` is no longer needed now that browsers support `fetch` natively.

Unfortunately in order to use native fetch, I needed to switch the tests over to run in a browser and not in Node -- which is probably good since this code is intended for the browser anyway. In order to do that, I needed to convert everything to ESM... so this ended up being a breaking change.

Once this merges, I'll be converting `frau-jwt` over to use this directly (and also become ESM) instead of `superagent` and `frau-superagent-xsrf-token`, and then subsequently upgrading BSI things that use it (there are 6).

### CODE REVIEW

**Review by omsmith — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -22,21 +22,19 @@
...
+    "eslint-config-brightspace": "^0.17",
+    "eslint-plugin-html": "^6",
+    "eslint-plugin-import": "^2",
+    "eslint-plugin-lit": "^1",
+    "eslint-plugin-sort-class-members": "^1",
+    "sinon": "^14"
+  },
+  "files": [
+    "/src"
+  ],
```
Comment (dlockhart) [inline package.json:38]: Replaced `.npmignore` with `files`.

File: package.json [hunk 2]
```diff
@@ -22,21 +22,19 @@
...
+  "exports": "./src/index.js"
```
Comment (dlockhart) [inline package.json:39]: This makes all the files other than the default one non-importable.

**Comment by dlockhart** [comment]
Investigating why the one set of local storage tests are failing in Firefox...

**Comment by omsmith** [comment]
> Investigating why the one set of local storage tests are failing in Firefox...

Wouldn't necessarily make a difference, but if you use a sinon sandbox then you don't need to store and reset every stub.

File: package.json [hunk 3]
```diff
@@ -22,21 +22,19 @@
...
-    "mocha": "^2.3.3",
-    "nock": "^2.15.0",
-    "sinon": "^1.17.2",
-    "sinon-chai": "^2.8.0",
-    "superagent-mock": "^1.7.0"
-  }
+    "@open-wc/testing": "^3",
+    "@web/test-runner": "^0.13",
+    "eslint": "^8",
+    "eslint-config-brightspace": "^0.17",
```
Comment (svanherk) [inline package.json]:
```suggestion
    "eslint-config-brightspace": "^0.18",
```
Or does this break a lot and you're leaving this for a separate upgrade?

Comment (dlockhart) [reply package.json]: Done. Haha wasn't intentional -- I just started these changes so long ago that the version changed in between.

**Comment by dlockhart** [comment]
> Wouldn't necessarily make a difference, but if you use a sinon sandbox then you don't need to store and reset every stub.

Yeah good call -- switched to using sandboxes everywhere.

**Review by svanherk — APPROVED** [review]

**Pull request merged**